src/components/Sidebar.tsx

```diff
@@ -22,129 +20,63 @@ const Sidebar = ({
   favs,
   map,
 }: {
-  reccs: User[] | undefined;
-  favs: User[] | undefined;
-  map: mapboxgl.Map | undefined;
+  reccs: User[];
+  favs: User[];
+  map: mapboxgl.Map;
 }) => {
-  const nurecs = requireNotUndefined(reccs);
-  const nufavs = requireNotUndefined(favs);
-
-  const [curList, setCurList] = useState<User[]>(nurecs);
-  const userToElem = (user: User) => {
-    return (
-      <div className="bg-stone-100 text-left px-2.5 py-2.5 rounded-xl m-3.5 align-center break-words">
-        <p className="font-bold">{user.name}</p>
-        <div className="flex flex-row space-x-4">
-          <div className="w-1/2">
-            <p>{user.startLocation}</p>
-            <p>{user.companyName}</p>
-            <button
-              onClick={() => viewRoute(user)}
-              className="underline decoration-dashed"
-            >
-              View Route
-            </button>
-          </div>
-          <div className="w-1/2">
-            {/* Add user bar */}
-            <div className="text-sm">
-              <p>{"Start: " + dateToTimeString(user.startTime)}</p>
-              <p>{"End: " + dateToTimeString(user.endTime)}</p>
-            </div>
-            <button className="bg-red-500 hover:bg-red-700 rounded-xl m-2 px-2 py-0.5 text-center text-white">
-              Connect
-            </button>
-          </div>
-        </div>
-      </div>
-    );
-  };
-
-  const viewRoute = (user: User) => {
-    if (map !== undefined) {
-      clearMarkers();
-
-      const startMarker = new mapboxgl.Marker({ color: "#2ae916" })
-        .setLngLat([user.startCoordLng, user.startCoordLat])
-        .addTo(map);
-
-      const endMarker = new mapboxgl.Marker({ color: "#f0220f" })
-        .setLngLat([user.companyCoordLng, user.companyCoordLat])
-        .addTo(map);
+  const [curList, setCurList] = useState<User[]>(reccs ?? []);
 
-      previousMarkers.push(startMarker);
-      previousMarkers.push(endMarker);
-
-      map.fitBounds([
-        [
-          Math.min(user.startCoordLng, user.companyCoordLng) - 0.125,
-          Math.max(user.startCoordLat, user.companyCoordLat) + 0.05,
-        ],
-        [
-          Math.max(user.startCoordLng, user.companyCoordLng) + 0.05,
-          Math.min(user.startCoordLat, user.companyCoordLat) - 0.05,
-        ],
-      ]);
-    }
-  };
+  useEffect(() => {
+    setCurList(reccs ?? []);
```

Comment:
Can you explain what this ternary is doing? How does it switch between recs vs favourites?